src/main/java/org/carlspring/cloud/storage/s3fs/util/Constants.java

```diff
@@ -0,0 +1,12 @@
+package org.carlspring.cloud.storage.s3fs.util;
+
+public final class Constants
+{
+
+    private Constants()
+    {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static final String S3_HOSTNAME = "s3.amazonaws.com";
```

Comment:
The use of virtual hosting style names is now the recommended standard https://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html

Reply:
@markjschreiber:
So, I guess virtual hosting style names are something like this:
```suggestion
    public static final String S3_HOSTNAME = "%s.s3.%s.amazonaws.com";
```

Where first argument should be the bucket name, and second argument the region, right?

So, in the method which uses this constant, how can I get the bucket name?
Example:

```
   /**
     * get the file system key represented by: the access key @ endpoint.
     * Example: access-key@s3.amazonaws.com
     * If uri host is empty then s3.amazonaws.com are used as host
     *
     * @param uri   URI with the endpoint
     * @param props with the access key property
     * @return String
     */
    protected String getFileSystemKey(URI uri, Properties props)
    {
        // we don`t use uri.getUserInfo and uri.getHost because secret key and access key have special chars
        // and dont return the correct strings
        String uriString = uri.toString().replace("s3://", "");
        String authority = null;

        int authoritySeparator = uriString.indexOf("@");

        if (authoritySeparator > 0)
        {
            authority = uriString.substring(0, authoritySeparator);
        }

        if (authority != null)
        {
            String host = uriString.substring(uriString.indexOf("@") + 1);

            int lastPath = host.indexOf("/");
            if (lastPath > -1)
            {
                host = host.substring(0, lastPath);
            }

            if (host.length() == 0)
            {
                final String bucketName = "";
                final String region = new DefaultAwsRegionProviderChain().getRegion().id();
                host = getDefaultHostName(bucketName, region);
            }

            return authority + "@" + host;
        }
        else
        {
            String accessKey = (String) props.get(ACCESS_KEY);
            accessKey = accessKey != null ? accessKey + "@" : "";

            String host = uri.getHost();
            if (host == null)
            {
                final String bucketName = "";
                final String region = new DefaultAwsRegionProviderChain().getRegion().id();
                host = getDefaultHostName("", region);
            }

            return accessKey + host;
        }
    }

    private String getDefaultHostName(final String bucketName, final String region)
    {
        return String.format(Constants.S3_HOSTNAME, bucketName, region);
    }
```

Reply:
The problem is that s3 doesn't have a root in terms of a URL that gives top level access to all of my buckets. There is no equivalent of `file:///`. In some ways it's more like windows where one bucket might be `C:` and another `D:` The method would need to be supplied a bucket name.